crates/ra_ide_api/src/completion/complete_postfix.rs

```diff
@@ -21,28 +22,46 @@ fn postfix_snippet(ctx: &CompletionContext, label: &str, detail: &str, snippet:
         .snippet_edit(edit)
 }
 
+fn is_bool_or_unknown(ctx: &CompletionContext) -> bool {
+    let receiver_ty = ctx.dot_receiver.and_then(|it| ctx.analyzer.type_of(ctx.db, it));
+    if let Some(ty) = receiver_ty {
+        match ty {
+            Ty::Apply(at) => match at.ctor {
+                TypeCtor::Bool => true,
+                _ => false,
+            },
+            Ty::Unknown => true,
+            _ => false,
```

Comment:
What about `Ty::Infer`?

Reply:
Mmm shouldn't it appear only _inside_ the type inference engine? 🤔

Reply:
Yeah, any inference variables that couldn't be resolved to an actual type would have been replaced by `Unknown` at this point.